<?xml version="1.0"?>

<launch>
  <arg name="Kp" default="30.0" doc="Proportional gain"/>
  <arg name="Ki" default="30.0" doc="Integrative gain"/>
  <arg name="Kd" default="51.0" doc="Derivative gain"/>
  <arg name="upper_limit" default="3000" doc="Upper limit"/>
  <arg name="lower_limit" default="-3000" doc="Lower limit"/>
  <arg name="windup_limit" default="1000" doc="Windup limit"/>
  <arg name="max_loop_frequency" default="30.0" doc="Max loop frequency"/>
  <arg name="min_loop_frequency" default="30.0" doc="Min loop frequency"/>

  <node name="controller2" pkg="driving_pid" type="controller2" ns="driving_pid/pid/motor1" output="screen" >
    <param name="node_name" value="driving_pid_motor1" />
    <param name="Kp" value="1.8" />
    <param name="Ki" value="4.5" />
    <param name="Kd" value="0" />
    <param name="upper_limit" value="$(arg upper_limit)" />
    <param name="lower_limit" value="$(arg lower_limit)" />
    <param name="windup_limit" value="$(arg windup_limit)" />
    <param name="max_loop_frequency" value="$(arg max_loop_frequency)" />
    <param name="min_loop_frequency" value="$(arg min_loop_frequency)" />
  </node>


  <node name="controller2" pkg="driving_pid" type="controller2" ns="driving_pid/pid/motor2" output="screen" >
    <param name="node_name" value="driving_pid_motor2" />
    <param name="Kp" value="1.8" />
    <param name="Ki" value="4.5" />
    <param name="Kd" value="0" />
    <param name="upper_limit" value="$(arg upper_limit)" />
    <param name="lower_limit" value="$(arg lower_limit)" />
    <param name="windup_limit" value="$(arg windup_limit)" />
    <param name="max_loop_frequency" value="$(arg max_loop_frequency)" />
    <param name="min_loop_frequency" value="$(arg min_loop_frequency)" />
  </node>


  <node name="controller2" pkg="driving_pid" type="controller2" ns="driving_pid/pid/motor3" output="screen" >
    <param name="node_name" value="driving_pid_motor3" />
    <param name="Kp" value="0.18" />
    <param name="Ki" value="0.45" />
    <param name="Kd" value="0.0" />
    <param name="upper_limit" value="$(arg upper_limit)" />
    <param name="lower_limit" value="$(arg lower_limit)" />
    <param name="windup_limit" value="$(arg windup_limit)" />
    <param name="max_loop_frequency" value="$(arg max_loop_frequency)" />
    <param name="min_loop_frequency" value="$(arg min_loop_frequency)" />
  </node>


  <node name="controller2" pkg="driving_pid" type="controller2" ns="driving_pid/pid/motor4" output="screen" >
    <param name="node_name" value="driving_pid_motor4" />
    <param name="Kp" value="0.18" />
    <param name="Ki" value="0.45" />
    <param name="Kd" value="0" />
    <param name="upper_limit" value="$(arg upper_limit)" />
    <param name="lower_limit" value="$(arg lower_limit)" />
    <param name="windup_limit" value="$(arg windup_limit)" />
    <param name="max_loop_frequency" value="$(arg max_loop_frequency)" />
    <param name="min_loop_frequency" value="$(arg min_loop_frequency)" />
  </node>


</launch>